MHVLib/Models/MHVModelBase.m

```diff
@@ -197,36 +197,44 @@ + (void)setProperties:(NSObject<MHVDataModelProtocol> *)object fromDictionary:(N
-                        if (propertyClass == [NSArray class] && !objectParameters)
+                        if (propertyClass)
                         {
-                            objectParameters = [MHVModelBase predictObjectClassFromPropertyName:propertyName];
+                            id value = nil;
+
+                            //objectParameters is array object type, or date formatter
+                            id objectParameters = propertyParametersCache[propertyName];
                             
-                            @synchronized (propertyParametersCache)
+                            //Not actually using this for anything yet, but could be useful...
+                            //Try to predict the array object type from the property name.  mapPoints -> MHVMapPoint
+                            if (propertyClass == [NSArray class] && !objectParameters)
                             {
-                                if (!objectParameters)
+                                objectParameters = [MHVModelBase predictObjectClassFromPropertyName:propertyName];
+
+                                @synchronized (propertyParametersCache)
                                 {
-                                    objectParameters = [NSString class];
+                                    if (!objectParameters)
+                                    {
+                                        objectParameters = [NSString class];
+                                    }
+
+                                    propertyParametersCache[propertyName] = objectParameters;
                                 }
+                            }
 
-                                propertyParametersCache[propertyName] = objectParameters;
+                            //All property type classes must implement, objectParameters ignored except for arrays and dates
+                            value = [[propertyClass alloc] initWithObject:dictionary[propertyNameInDictionary]
+                                                         objectParameters:objectParameters];
+                            if (value)
+                            {
+                                [object setValue:value forKey:propertyName];
                             }
                         }
-                        
-                        //All property type classes must implement, objectParameters ignored except for arrays and dates
-                        value = [[propertyClass alloc] initWithObject:dictionary[propertyNameInDictionary]
-                                                     objectParameters:objectParameters];
-                        if (value)
+                        else
                         {
-                            [object setValue:value forKey:propertyName];
+                            // for base types (int, long, etc)
+                            [object setValue:dictionary[propertyNameInDictionary] forKey:propertyName];
```

Comment:
This one is interesting. Without this change, the json deserializer doesn't deserialize to base objects like longs. That's a problem for MHVTime where the minutes and hours are stored as ints. This approach made the most sense to me. Otherwise, we could change the MHVTime to use NSNumber, though that would be a more far-reaching change.